### ISSUE TITLE
Extract comments in header files

### ISSUE THREAD

**Issue by ngocdaothanh** [CastXML/CastXML#152]
Hi, is there a way to also extract source code comments in C++ header files into the output XML file?

I'd like to create doc from header files.

**Comment by bradking** [comment]
No, CastXML is only meant to provide the API.

Extracting comments is much easier because one does not need to have a whole translation unit (preprocessed source file).  There are many tools meant for that, such as [Doxygen](http://www.doxygen.nl/).

**Comment by todoooo** [comment]
Could it not be added with a command line option? It would be useful to have both the API and documentation in a single XML file for post-processing.

**Comment by ngocdaothanh** [comment]
It want to do processing on the doc and methods/classes at the same time (example: when creating Swift language wrapper for C/C++ libraries), so it's very useful if CastXML can also provide the doc.

> Extracting comments is much easier

I'm new to Clang and CastXML, but if it's easy, please instruct me how to do.
I'll create a PR for this feature.

**Comment by todoooo** [comment]
There are some clues on how to do it here https://github.com/CastXML/CastXML/issues/117

**Comment by Smattr** [comment]
A bit late to the party, but to add my own perspective on @bradking's point, you don't need a full C/C++ parser for comment extraction. You just need a parser capable of recognising comments.

However, I can certainly appreciate your use case of wanting to consume a single XML document that contains both the code and comments. But it's not clear what such a thing would look like. IMHO there's no neat way to model comments in XML because XML is inherently hierarchical and C/C++ comments can appear in arbitrary positions.

I am only aware of two reasonable strategies for this. Something like [srcML](https://www.srcml.org/) where the original literal text is retained within the XML or, as Brad said, something like Doxygen where your comments have a structured form that mean something outside of the C/C++ standards. Otherwise you're stuck with something like Clang's approach, where your parser has to make opinionated decisions about which AST node to attach a comment to.

**Comment by EricCousineau-TRI** [comment]
I think it's totally merited to have `CastXML` produce comments in some form.

For us in Drake, when we generate bindings, we generally want to ensure we have correspondence between the C++ symbol and the Python bindings. Sure, we can generate whatevs with Doxygen or a separate tool, but then using those comments from the bindings is a pain.

For the record, we currently use `clang.cindex`, which does provide the `raw_comment`, and we've found it very useful for generating docstrings.

I opened #172, but am closing that since it's a duplicate of this issue. From that issue:

> `libclang` exposes `cursor.raw_comment`, which I guess comes from here from `clang`:
> https://clang.llvm.org/doxygen/classclang_1_1RawComment.html
> 
> It would be nice to access this as an option?
